agents-api/agents_api/common/retry_policies.py

```diff
@@ -0,0 +1,18 @@
+from datetime import timedelta
+
+from temporalio.common import RetryPolicy
+
+DEFAULT_RETRY_POLICY = RetryPolicy(
+    initial_interval=timedelta(seconds=1),
+    backoff_coefficient=2,
+    maximum_attempts=2,
+    maximum_interval=timedelta(seconds=10),
+    non_retryable_error_types=[
+        "WorkflowExecutionAlreadyStarted",
+        "TypeError",
+        "AssertionError",
+        "HTTPException",
+        "SyntaxError",
+        "ValueError",
```

Comment:
we should Jinja template error, jsonschema validation error, pydantic validation error, zero division error and others (from chatgpt):

asyncio.CancelledError
Description: Raised when an asynchronous task is canceled.
Reason Not to Retry: The task has been explicitly canceled and should not continue.
asyncio.InvalidStateError
Description: Indicates that an operation is invalid in the current state.
Reason Not to Retry: The current state prevents the operation from succeeding.
9. requests
requests.exceptions.InvalidURL
Description: Raised when the URL provided is malformed.
Reason Not to Retry: The URL must be corrected before making the request again.
requests.exceptions.MissingSchema

temporalio
temporalio.exceptions.TerminalFailure
Description: Indicates that a workflow has reached a terminal state.
Reason Not to Retry: The workflow cannot proceed further.
temporalio.exceptions.CanceledError
Description: Raised when a workflow is canceled.
Reason Not to Retry: The workflow has been explicitly canceled and should not continue.
3. fastapi
fastapi.exceptions.HTTPException
Description: Represents HTTP errors like 400 Bad Request or 404 Not Found.
Reason Not to Retry: Client-side errors unlikely to succeed upon retrying without changes.
fastapi.exceptions.RequestValidationError
Description: Occurs when request data fails validation.
Reason Not to Retry: Invalid input data needs correction before retrying.
4. jinja2
jinja2.exceptions.TemplateSyntaxError
Description: Raised for syntax errors in templates.
Reason Not to Retry: Template needs correction; retrying won't fix syntax issues.
jinja2.exceptions.TemplateNotFound
Description: Occurs when a specified template cannot be found.
Reason Not to Retry: The template must be added or the path corrected before retrying.
5. httpx
httpx.RequestError
Description: Represents network-related errors, such as DNS failures or connection timeouts.
Reason Not to Retry: Some network issues are persistent and won't resolve with retries.
httpx.HTTPStatusError
Description: Raised for non-success HTTP status codes like 400 or 500.
Reason Not to Retry: Client or server errors that require different handling strategies.
6. pydantic
pydantic.ValidationError